### ISSUE TITLE
Install on Raspberry Pi 3 model B - raspbian Jesse Lite

### ISSUE THREAD

**Issue by jwhittin744** [snhirsch/katana-midi-bridge#13]
The install seems to go fine, but when i follow the steps through to the "Show_Interfaces" command, the last line beginning with 'USB' is missing from the output as shown here...  Because of this, I cannot update the Start file for the midi tokens...

/katana-midi-bridge $ sudo ./show_interfaces
Output Ports

KATANA:KATANA MIDI 1 20:0
KATANA:KATANA MIDI 2 20:1
KATANA:KATANA MIDI 3 20:2
Midi Through:Midi Through Port-0 14:0

Input Ports

KATANA:KATANA MIDI 1 20:0
KATANA:KATANA MIDI 2 20:1
KATANA:KATANA MIDI 3 20:2
Midi Through:Midi Through Port-0 14:0

any ideas?

**Comment by snhirsch** [comment]
It appears that your MIDI controller is either not connected to the Pi or not being recognized.  What make and model of controller are you using?

**Comment by jwhittin744** [comment]
Thanx for the reply.


Controller is Behringer FCB1010,


USB-Midi interface is M-audio Uno

[cid:1031bbc7-840d-45f2-ade0-35ec62806794]

The interface is working (in general) as I use the FCB1010 Editor software on my laptop to program the controller.

Is there a better interface that you would recommend?  would it be related to the FCB1010 firmware?

For what it's worth, I'm having issues getting it to work with GumTown's FX Floorboard as well...

Thanx,

Jeff W.
________________________________

**Comment by snhirsch** [comment]
Hmm.  If you tail the system log file on the Pi, do you see **anything** happening when you plug the M-audio Uno in?  Also try running 'lsusb' to see if the Uno device is even being enumerated.

It really does sound as if the USB/MIDI adapter is not being properly recognized.  

I've used an M-Audio MidiSport 1x1 and had no compatibility issues ever.

**Comment by jwhittin744** [comment]
here's the output of the lsusb and show_interfaces command.  it does detect the Midi-USB interface on usb port 4...


jwhittington@raspberrypi:~/katana-midi-bridge $ sudo ./show_interfaces
Output Ports

KATANA:KATANA MIDI 1 20:0
KATANA:KATANA MIDI 2 20:1
KATANA:KATANA MIDI 3 20:2
Midi Through:Midi Through Port-0 14:0

Input Ports

KATANA:KATANA MIDI 1 20:0
KATANA:KATANA MIDI 2 20:1
KATANA:KATANA MIDI 3 20:2
Midi Through:Midi Through Port-0 14:0

jwhittington@raspberrypi:~/katana-midi-bridge $ lsusb
Bus 001 Device 005: ID 0582:01d8 Roland Corp.
Bus 001 Device 004: ID 0763:1010 Midiman MidiSport 1x1
Bus 001 Device 003: ID 0424:ec00 Standard Microsystems Corp. SMSC9512/9514 Fast Ethernet Adapter
Bus 001 Device 002: ID 0424:9514 Standard Microsystems Corp.
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
jwhittington@raspberrypi:~/katana-midi-bridge $

Tail of the syslog shows this... I disconnected the Midi-USB interface then reconnected it.

Mar  6 23:36:54 raspberrypi kernel: [ 1989.934847] usb 1-1.2: USB disconnect, device number 4
Mar  6 23:36:55 raspberrypi dhcpcd[813]: eth0: fe80::5a6d:8fff:fe0d:374 router available
Mar  6 23:37:02 raspberrypi kernel: [ 1997.655607] usb 1-1.2: new full-speed USB device number 6 using dwc_otg
Mar  6 23:37:02 raspberrypi kernel: [ 1997.786699] usb 1-1.2: New USB device found, idVendor=0763, idProduct=1010
Mar  6 23:37:02 raspberrypi kernel: [ 1997.786712] usb 1-1.2: New USB device strings: Mfr=0, Product=0, SerialNumber=0
Mar  6 23:37:02 raspberrypi /katana_bridge_start: 2441: Cannot find controller MIDI device
Mar  6 23:37:03 raspberrypi /katana_bridge_start: 2453: Cannot find controller MIDI device


Thanx,

Jeff W



________________________________

**Comment by snhirsch** [comment]
Wait, you said M-Audio Uno, but your log calls out a Midisport 1x1 (the same USB/MIDI interface I'm using).  Which hardware are you using?  Are you sure you have all the prerequisite libraries installed on the Pi?  ALSA is supposed to identify it and bind it to a MIDI device.  A couple more things:

1. Did you add the command flag from my instructions to force "full-speed" operation?  If so, try without it.  If not, please add it.

2. Try uninstalling the katana bridge in case the failed startup is interfering with ALSA in some manner. 

3. Are you using a USB hub?

4. Does 'aconnect -l' show you the Midisport?

**Comment by jwhittin744** [comment]
I know - that's weird about how it's showing the interface - the Midisport is the little box that doesn't have any cables, isn't it?  Still an M-Audio product, though, right?  I wonder if they're the same board, just that the Uno has the cables and the Midisport doesn't?


I did follow your wiki from top to bottom - I can re-run the installs just to make sure.  No usb hub - the katana and the uno are connected directly to USB ports on the Pi (the Pi3 model B has 4 USB ports).


I'll try steps 1, 2 and 4 tonight after work.


I appreciate all the help here - thanx!



________________________________

**Comment by jwhittin744** [comment]
update..


I ran aconnect -l and didn't see the midi interface so I removed the setting to force the usb to high speed.


rebooted and this is the output from aconnect -l


jwhittington@raspberrypi:~ $ sudo aconnect -l
[sudo] password for jwhittington:
client 0: 'System' [type=kernel]
    0 'Timer           '
    1 'Announce        '
client 14: 'Midi Through' [type=kernel]
    0 'Midi Through Port-0'
client 20: 'KATANA' [type=kernel]
    0 'KATANA MIDI 1   '
    1 'KATANA MIDI 2   '
    2 'KATANA MIDI 3   '

uninstalled the bridge, and no difference...


________________________________

**Comment by snhirsch** [comment]
What about the boot command line parameter to setup USB speed? - Sorry missed your comment about the high-speed setting.

I'm about out of ideas.  Can you tell me which OS and version you're running on the Pi and which Pi model you have?  I guess it's possible your USB adapter is not MIDI class-compliant.  The ALSA layer will automatically recognize any interface that's class-compliant and has been extended to do this also for Roland/Boss products (which are NOT class-compliant).  Since the Katana **is** being recognized, I think it's down to your USB/MIDI adapter.  A Roland UM-1 or Midisport 1x1 should work properly (even though yours ids as a midisport, I think it's different from what I have here).

**Comment by jwhittin744** [comment]
Yes, I removed the boot command line parameter ...


I'm wondering if it's that the interface I have is too old?  I've had it for about 10 years.....


________________________________

**Comment by snhirsch** [comment]
That could possibly predate the current "class compliance" specification.  May be worth sending a note to M-Audio to see if there's a firmware upgrade, or if they can shed any light on why it's not recognized by a contemporary Linux system.

Or, go on Amazon and purchase a new Midisport 1x1 and be done with it :-)

**Comment by jwhittin744** [comment]
Yeah, I think I'm going to take this as an opportunity to hit my favorite music stores and pick up a new interface - likely the Roland UM-One Mk2 (as long as it says it's Class Compliant)....


There are no firmware updates for the M-Audio UNO and no specific drivers for it for Linux either.


Thanx


Jeff W.


________________________________

**Comment by jwhittin744** [comment]
ok - so getting a new USB-Midi interface has solved the problem.


Thanx for your help!


JW


________________________________

**Comment by snhirsch** [comment]
Excellent!  Glad you were able to sort it out.  I hope you find the bridge software useful.  I had originally planned to expose the entire sysex API, but ran out of time and motivation.  A few folks have found the snapshot feature to be useful, though.

I'm hoping someone with Python coding skills can pickup the API work.
